src/corehost/src/main.cpp

```diff
@@ -8,30 +8,53 @@
 #include "utils.h"
 #include "coreclr.h"
 
-void get_tpafile_path(const pal::string_t& app_base, const pal::string_t& app_name, pal::string_t& tpapath)
+void get_tpafile_path(const pal::string_t& app_base, const pal::string_t& app_name, pal::string_t* tpapath)
 {
-    tpapath.reserve(app_base.length() + app_name.length() + 5);
+    tpapath->reserve(app_base.length() + app_name.length() + 5);
```

Comment:
`pal::string_t` is just a typedef for `std::string`/`std::wstring` depending on platform so it just depends on how that is implemented.